src/NBitcoin/ConcurrentChain.cs

```diff
@@ -26,15 +26,16 @@ public ConcurrentChain()
 
         public ConcurrentChain(BlockHeader genesisHeader)
         {
+            // TODO: Pass NetworkOptions to GetHash
```

Comment:
Ah perhaps we can pass Network in the constructor of ConcurrentChain (or ChainBase) that will solve all our problems.

Reply:
Added.